### PR TITLE
Add Range method to iterate through Value tree

### PR DESCRIPTION
**Pull request by phirework**
This adds a Range function for walking the AST depth-first and applying a custom function to each leaf

To be used in tailscale/corp#8412

### CODE REVIEW

File: format.go [hunk 1]
```diff
@@ -88,6 +100,25 @@ func (v *Value) Format() {
...
+		for i := 0; i < v2.length(); i++ {
+			v2.valueAt(i).Range(f)
+		}
+	}
+}
+
+// SetValue mutates the value.
+func (v *Value) SetValue(vt ValueTrimmed) {
+	v.Value = vt
+}
```
Comment (bradfitz) [inline format.go:120]: this is assigning only an exported field. Any reason for the method? Is it satisfying some interface, or?

**Comment by dsnet** [comment]
BTW, the hujson repository has historically used a different title convention. Instead of "package: some description", it seems to use a sentence. For example: "Add Range method to iterate through a Value tree"

File: format.go (hunk 1 above)
Comment (phirework) [reply format.go]: Javascript habits 😅 removed

**Review by dsnet — APPROVED** [review]
Thanks!

**Comment by phirework** [comment]
Thank you!

**Pull request merged**